### ISSUE TITLE
[Android] Orientation changed event fired before OS rotates

### ISSUE THREAD

**Issue by rogarMS** [wonday/react-native-orientation-locker#45]
I haven't test on iOS devices, but have on Android.

Both events addOrientationListener and addDeviceOrientationListener seem to fire before the OS actually rotates the screen. 
Ex. Hold the phone upright portrait and slowly rotate about 45deg towards landscape. The event will fire for landscape while the phone/app is still portrait. Continue rotating and then the OS will finally rotate.
It's not a race condition because you can hold it at that position and stay there. It seem's like its based on the gyro alone and not what the device/OS thinks.

This is causing some where situations where the JS thinks it's landscape, but the device is still portrait, causing weird layout issues. 

react-native-orientation didn't have this issue.

**Comment by gp3gp3gp3** [comment]
I've confirmed this issue happens for iOS as well.  It's the main reason I had to remove it from my project.

For now the best approach I've found is using `react-native-orientation` for locking/unlocking the orientation, and using an event listener on the React Native `Dimensions` API to detect orientation change.  If you're using state management, use those values in the event listener instead of making a call to `Dimensions.get` so you don't need to make another call over the native bridge.
